packages/dds/tree/src/simple-tree/api/configuration.ts

```diff
@@ -345,15 +372,55 @@ export function checkUnion(
 		);
 	}
 
+	if (records.length > 1) {
+		ambiguityErrors.push(
+			`More than one kind of record allowed within union (${formatTypes(records)}). This would require type disambiguation which is not supported by records during import or export.`,
+		);
+	}
+
 	if (maps.length > 0 && arrays.length > 0) {
 		ambiguityErrors.push(
 			`Both a map and an array allowed within union (${formatTypes([...arrays, ...maps])}). Both can be implicitly constructed from iterables like arrays, which are ambiguous when the array is empty.`,
 		);
 	}
 
-	if (objects.length > 0 && maps.length > 0) {
+	/**
+	 * Constructs a sentence-formatted list of entries, using commas and "and" as appropriate.
+	 */
+	function sentenceList(entries: readonly string[]): string {
+		switch (entries.length) {
+			case 0: {
+				return "";
+			}
+			case 1: {
+				return entries[0] ?? oob();
+			}
+			case 2: {
+				return `${entries[0] ?? oob()} and ${entries[1] ?? oob()}`;
+			}
+			default: {
+				return `${entries.slice(0, -1).join(", ")}, and ${entries[entries.length - 1]}`;
+			}
+		}
+	}
+
+	if (
+		(objects.length > 0 ? 1 : 0) + (maps.length > 0 ? 1 : 0) + (records.length > 0 ? 1 : 0) >
+		1
+	) {
+		const nodeKindListEntries = [];
+		if (objects.length > 0) {
+			nodeKindListEntries.push("objects");
+		}
+		if (maps.length > 0) {
+			nodeKindListEntries.push("maps");
+		}
+		if (records.length > 0) {
+			nodeKindListEntries.push("records");
+		}
```

Comment:
There is some redundant checking here.

I think having` nodeKindListEntries` renamed to `nodeKindListEntriesFromObjects`, then replacing the ```if (
		(objects.length > 0 ? 1 : 0) + (maps.length > 0 ? 1 : 0) + (records.length > 0 ? 1 : 0) >
		1
	)``` above with a `if (nodeKindListEntries.length > 1)` below would be better.